multiqc/modules/megahit/megahit.py

```diff
@@ -0,0 +1,146 @@
+
+from multiqc import config
+from multiqc.modules.base_module import BaseMultiqcModule, ModuleNoSamplesFound
+from multiqc.plots import table
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """MEGAHIT module"""
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="MEGAHIT",
+            anchor="megahit",
+            href="https://github.com/voutcn/megahit",
+            info="is an ultra-fast and memory-efficient NGS assembler",
+            doi="10.1093/bioinformatics/btv033",
+        )
+
+        data = dict()
+        for f in self.find_log_files("megahit", filehandles=True):
+            for line in f["f"]:
+                self.add_data_source(f, f["s_name"])
+                if " - MEGAHIT v" in line:
+                    version = line.split(" - MEGAHIT v")[1].strip()
+                    self.add_software_version(version, sample=f["s_name"])
+                elif " - Memory used: " in line:
+                    mem = line.split(" - Memory used: ")[1].strip()
+                    data[f["s_name"]] = {"megahit_mem": mem}
+                elif " - ALL DONE. Time elapsed: " in line:
+                    time = line.split(" - ALL DONE. Time elapsed: ")[1].strip().replace(" seconds", "")
+                    data[f["s_name"]]["megahit_time"] = time
+                elif " contigs, total " in line:
+                    # Parse line like this:
+                    # 2022-06-13 14:22:47 - 64408 contigs, total 46322050 bp, min 200 bp, max 94571 bp, avg 719 bp, N50 831 bp
+                    regex = re.compile(
+                        r"(\d+) contigs, total (\d+) bp, min (\d+) bp, max (\d+) bp, avg (\d+) bp, N50 (\d+) bp"
+                    )
+                    match = regex.search(line)
+                    if match:
+                        data[f["s_name"]]["megahit_contigs"] = int(match.group(1))
+                        data[f["s_name"]]["megahit_bases"] = int(match.group(2))
+                        data[f["s_name"]]["megahit_min_contig"] = int(match.group(3))
+                        data[f["s_name"]]["megahit_max_contig"] = int(match.group(4))
+                        data[f["s_name"]]["megahit_avg_contig"] = int(match.group(5))
+                        data[f["s_name"]]["megahit_n50"] = int(match.group(6))
+
+        # Filter to strip out ignored sample names
+        data = self.ignore_samples(data)
+        if len(data) == 0:
+            raise ModuleNoSamplesFound
+        log.info(f"Found {len(data)} reports")
+
+        # Write parsed report data to a file
+        self.write_data_file(data, f"multiqc_{self.anchor}")
+
+        headers = {
+            "megahit_mem": {
+                "title": "Memory",
+                "description": "Memory used",
+                "min": 0,
+                "scale": "Reds",
+                "format": lambda x: humanize.naturalsize(x),
```

Comment:
I have a feeling that this could give different units for different rows, right? I think it's better to pick a single unit for all samples, so that the numbers can be compared easily.

eg. it's very tough to spot which of the following is bigger at a glance:

13.3 GB
479.2 GB
4.3 TB
7.4 GB
933.2 MB

etc..

Reply:
Fair enough. Perhaps `humanize` is not a good fit for MultiQC after all :) Can switch to a more classical approach of reporting a decimal group separated number of KB